routers/api/v1/repo/issue_label.go

```diff
@@ -351,8 +351,8 @@ func prepareForReplaceOrAdd(ctx *context.APIContext, form api.IssueLabelsOption)
 	}
 
 	if !ctx.Repo.CanWriteIssuesOrPulls(issue.IsPull) {
```

Comment:
Maybe we could check permissions before querying the labels from db, so that we can return immediately if the doer doesn't have permissions.

Reply:
8e7f3669ff76675a38daa2242223c1da5067be4f